src/App.tsx

```diff
@@ -193,6 +193,54 @@ function App() {
   const [activeAiPatchContainerId, setActiveAiPatchContainerId] = useState<string | null>(null);
   const [activeAiSubMaskId, setActiveAiSubMaskId] = useState<string | null>(null);
   const [zoom, setZoom] = useState(1);
+  const [displaySize, setDisplaySize] = useState({ width: 0, height: 0 });
+  const [previewSize, setPreviewSize] = useState({ width: 0, height: 0 });
+  const [baseRenderSize, setBaseRenderSize] = useState({ width: 0, height: 0 });
+  const [originalSize, setOriginalSize] = useState({ width: 0, height: 0 });
+  const [isFullResolution, setIsFullResolution] = useState(false);
+  const [fullResolutionUrl, setFullResolutionUrl] = useState<string | null>(null);
+  const [isLoadingFullRes, setIsLoadingFullRes] = useState(false);
+  const fullResRequestRef = useRef<any>(null);
+
+  // Track display size changes for zoom calculations
+  const handleDisplaySizeChange = useCallback((size: any) => {
```

Comment:
You should use `ImageDimensions` interface here with `size`, imported from `useImageRenderSize.tsx`. Then you can modify the `useState`'s to `useState<ImageDImensions>({ width: 0, height: 0 });` etc.